### ISSUE TITLE
Connect status history page to status bucket

### ISSUE THREAD

**Issue by ebb-tide** [influxdata/influxdb#14548]
needs: create swagger definition and confirm before implementing. 
get check history for all checks, or get check history for check ID should be possible to implement. Needs to accept time period, checkID, tag key/value pairs as parameters. (perhaps it should just accept a flux query?)
get check history page will use this API
Also "preview using historical data" will use this API

**Comment by chnn** [comment]
I think we'll want to supply the following parameters to the status API endpoint:

- `limit`
- `offset`
- `sortBy` (time, status, checkID, message)
- `timeRange` (absolute start/stop as unix times)

In addition I think we will want to be able to filter results, either by `status`, `checkID`, `message`, or by the additional user tags written to the status. For any of these tags, we will want to be able to filter using `==`, `!=`, `=~`, or `!~` as the operator.

Since this is complicated and we can already do this in Flux, I think we should deprioritize implementing this endpoint for now.

**Comment by ebb-tide** [comment]
Since user has unrestricted access to data system buckets, we may no longer need this endpoint. 
In that case this issue should change to UI generating the appropriate flux query to power status page.

**Comment by chnn** [comment]
I tested out powering the the alert history viewer with Flux, using syslog data as a substitute. Here's the query I used:

```
// Utils
import {runQuery} from 'src/shared/apis/query'

// Types
import {LoadRowsOptions} from 'src/eventViewer/types'

export const runSyslogQuery = ({offset, limit, since}: LoadRowsOptions) => {
  const query = `
from(bucket: "telegraf")
  |> range(start: -30d, stop: ${Math.round(since / 1000)})
  |> filter(fn: (r) => r._measurement == "syslog" and r._field == "message")
  |> group()
  |> keep(columns: ["_time", "_value", "appname", "facility", "host", "severity"])
  |> rename(columns: {"_time": "time", "_value": "message"})
  |> limit(n: ${limit}, offset: ${offset})
`

  return runQuery(ORG_ID, query)
}
```

It worked well. I think the important parts are:

- Query should return a single table
- The definition of `now()` needs to be pinned to a specific time, so that the meaning of the `limit` + `offset` pair doesn't change each time the query is run

Left out is support for the `LoadRowsOptions['filter']` argument (searching support), but I think that should straightforward.

**Comment by chnn** [comment]
There are also some questions I have for the Flux team around performance.

Mainly I want to know if:

1. Query one gazillion rows
2. Regex
3. Limit to 10 rows

Is as fast as:

1. Query one gazillion rows
2. Limit to 10 rows
3. Regex

This affects what we sort of search feature we support.

**Comment by chnn** [comment]
Currently blocked on the existence of a system status bucket.

**Comment by chnn** [comment]
Status bucket exists, but there's no statuses in it. Might be most efficient to wait until there is before hooking the bucket to the UI.

**Comment by chnn** [comment]
Also came across another issue. Here's the status history page:

![table](https://gist.githubusercontent.com/chnn/62bdb41eaab5a8072800d0ae4d7783c7/raw/765e9f5422f9e0b12470634f8f4de8caf080959f/the%2520table.png)

It contains a column of all the user-written tags for each status:

![tags column](https://gist.github.com/chnn/62bdb41eaab5a8072800d0ae4d7783c7/raw/2878c9bc66854b90b717d5d9d2ad46a2008bf130/tags%2520column.png)

We do not want to load all the statuses at once for this page, so we use the `limit` function in Flux to implement pagination. A simplified version of the Flux query we use looks like:

```
from(bucket: "000000000000000b")
  |> range(start: -60d, stop: t1)
  |> filter(fn: (r) => r._measurement == "statuses")
  |> keep(columns: ["_time", "_check_id", "_check_name", "_level", "_message"])
  |> group()
  |> limit(n: n, offset: k)
```

(`n`, `k`, and `t1` are parameters supplied by the UI.)

This query works well. However, it doesn't return the user tags on a status. If we omit the `keep` call to get the user tags, then the ungrouping operation may break (`|> group()`), since it could combine multiple conflicting schemas and panic. The ungrouping operation is necessary for the limit + offset pagination to make sense.

Here are the solutions I can think of:

- Remove the tags column from the status histoy
- Provide a Flux solution to the conflicting schema issue

We have [had a similar issue in the past](https://github.com/influxdata/flux/issues/1298#issuecomment-510536882), and solved it client-side by spreading multiple schemas into a single table in a safe way. But we need to do that processing in Flux now, so that we can have pagination

**Comment by chnn** [comment]
Just discussed this issue with @pauldix and @nathanielc. For the immediate issue at hand, we will remove the tags column from the status history page. Since the message for each status will likely include relevant tag data, we think this is a good compromise for the alerting MVP.

Long term, we are looking into changes to Flux which will allow combining heterogeneous groups into a single table, which will enable the `limit` function to work for us. See https://github.com/influxdata/flux/issues/439.

The remaining blocker for completing this issue is having real statuses in the system status bucket to query + verify with.
